packages/rum-core/src/browser/domMutationObservable.ts

```diff
@@ -1,17 +1,12 @@
-import { monitor, Subscription } from '@datadog/browser-core'
+import { monitor, Observable } from '@datadog/browser-core'
 
-export interface DOMMutationObservable {
-  subscribe(callback: () => void): Subscription
-}
-
-export function createDOMMutationObservable(): DOMMutationObservable {
-  let callbacks: Array<() => void> = []
+export function createDOMMutationObservable() {
   const MutationObserver = getMutationObserverConstructor()
-  const observer = MutationObserver ? new MutationObserver(monitor(notify)) : undefined
-
-  function notify() {
-    callbacks.forEach((callback) => callback())
-  }
+  const observable = new Observable<void>(() => {
+    startDOMObservation()
+    return stopDOMObservation
+  })
```

Comment:
I think this could be simpler written as:

```suggestion
  const observable = new Observable<void>(() => {
    if (!MutationObserver) return
    
    const observer = new MutationObserver(monitor(() => observable.notify()))
 
    observer.observe(document, {
      attributes: true,
      characterData: true,
      childList: true,
      subtree: true,
    })

    return () => observer.disconnect()
  })
```

It avoids a maybe undefined observer.